agentos/core.py

```diff
@@ -45,14 +52,58 @@ class Agent(MemberInitializer):
     learning, use of models, state updates, etc.
     """
 
-    def learn(self):
-        """Does one iteration of training"""
-        pass
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.curr_obs = None
+        self._should_reset = True
+
+    def step(self):
```

Comment:
> Can you say more about why you pushed the stepping/rollout logic from the runtime into the agent?

My thinking was that rolling out an episode is a pretty standard thing to do with an Agent and that running a rollout requires access to all the components within the Agent (i.e. Environment, Policy, Trainer, Dataset), thus it made sense to put it as a method on Agent.

A couple of thoughts while I'm digging around here:
* It seems odd for Agent to contain (or, at least, have a pointer to) the Environment given the standard textbook depiction of the Agent/Environment loop.  This might be confusing to people coming to AOS for the first time. Is this what motivated you to try renaming Agent to Composer?  No solution from me at the moment but it did strike me as I was looking at the code again.
* Looking at some of your example agents, I can see how a separate `rollout()` can also be useful (e.g. for an evolutionary agent evaluating multiple policies).  Maybe when we port forward the example agents, we reconstitute a top-level rollout and then `Agent.rollout` becomes a wrapper around it?

> Also why did you chose the name step() for this?

Renaming to `advance()` as per my comment [here](https://github.com/agentos-project/agentos/pull/134#discussion_r703046444).

Reply:
:+1: giving #136 a look now!